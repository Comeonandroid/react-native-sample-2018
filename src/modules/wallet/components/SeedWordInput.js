import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import Icon from '../../../utils/icon'

import { View } from '../../../elements'
import { SectionHeader } from '../../../elements'
import { Textarea } from '../../../elements'

class SeedWordInput extends PureComponent {
  render() {
    const { header, onChange, onBlur, seed_word, disabled } = this.props
    return (
      <Wrapper>
        <Icon name='password-with-asterics' size={18} />
        <WalletHeader>{header}</WalletHeader>
        <Seedarea 
          onChange={onChange} 
          onBlur={onBlur} 
          name="seed_word" 
          value={seed_word} 
          editable={!disabled}
        />
      </Wrapper>
    )
  }
}

SeedWordInput.propTypes = {
  seed_word: PropTypes.string,
  header: PropTypes.string,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  disabled: PropTypes.bool
}

SeedWordInput.defaultProps = {
  seed_word: ''
}

const Wrapper = styled(View) `

`

const WalletHeader = styled(SectionHeader) `
  color: ${ ({ theme }) => theme.colors.primary}
`

const Seedarea = styled(Textarea) `
  margin: 20px 0px;
  padding-top: 22px;
`

export default SeedWordInput
