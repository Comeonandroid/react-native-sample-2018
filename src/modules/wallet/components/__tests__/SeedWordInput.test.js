import 'react-native'
import React from 'react'
import { shallow } from 'enzyme'

import { itBehavesLike } from '../../../../specs/shared_examples/index'

import SeedWordInput from '../SeedWordInput'

const setup = propOverrides => {
  const props = Object.assign({
    header: 'test',
    seed_word: 'test word',
    onChange: jest.fn()
  }, propOverrides)

  const component = <SeedWordInput {...props} />
  const wrapper = shallow(component)

  return {
    props,
    wrapper,
    component
  }
}

describe('<SeedWordInput />', () => {
  const { props, component, wrapper } = setup()
  itBehavesLike('aThemedComponent', { subject: component })

  it('properly fires `onChange` when input changes', () => {
    const input = wrapper.find({ name: 'seed_word' })
    const event = { target: { value: 'Here is a value' } }
    input.simulate('change', event)
    expect(props.onChange).toHaveBeenCalledWith(event)
  })

  it('pass val to input', () => {
    const input = wrapper.find({ name: 'seed_word' })
    expect(input.props().value).toBe(props.seed_word)
  })

  it('header to input', () => {
    const header = wrapper.children().get(1)
    expect(header.props.children).toBe(props.header)
  })
})
