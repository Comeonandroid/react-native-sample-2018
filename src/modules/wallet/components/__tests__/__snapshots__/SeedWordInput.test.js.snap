// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeedWordInput /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {},
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 18,
        },
        Array [
          Object {
            "color": "#fcfcfc",
          },
          undefined,
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fcfcfc",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "fontFamily": "Lato",
            "fontSize": 15,
            "fontWeight": "400",
            "marginBottom": 5,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 5,
          },
          Array [
            Object {
              "color": "#faa632",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    test
  </Text>
  <TextInput
    allowFontScaling={true}
    editable={true}
    error={false}
    height={120}
    multiline={true}
    name="seed_word"
    numberOfLines={10}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onChangeText={[Function]}
    placeholder={undefined}
    placeholderTextColor="#c1c1c1"
    style={
      Array [
        Object {
          "backgroundColor": "#282828",
          "borderColor": "#282828",
          "borderWidth": 1,
          "color": "#fcfcfc",
          "fontSize": 14,
          "fontStyle": "italic",
          "minHeight": 120,
          "paddingBottom": 8,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 8,
        },
        Array [
          Object {
            "marginBottom": 20,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 20,
            "paddingTop": 22,
          },
          undefined,
        ],
      ]
    }
    value="test word"
  />
</View>
`;

exports[`<SeedWordInput /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {},
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 18,
        },
        Array [
          Object {
            "color": "#4b4c4d",
          },
          undefined,
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "fontFamily": "Lato",
            "fontSize": 15,
            "fontWeight": "400",
            "marginBottom": 5,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 5,
          },
          Array [
            Object {
              "color": "#faa632",
            },
            undefined,
          ],
        ],
      ]
    }
  >
    test
  </Text>
  <TextInput
    allowFontScaling={true}
    editable={true}
    error={false}
    height={120}
    multiline={true}
    name="seed_word"
    numberOfLines={10}
    onBlur={[Function]}
    onChange={[MockFunction]}
    onChangeText={[Function]}
    placeholder={undefined}
    placeholderTextColor="#c1c1c1"
    style={
      Array [
        Object {
          "backgroundColor": "#5d5959",
          "borderColor": "#5d5959",
          "borderWidth": 1,
          "color": "#fcfcfc",
          "fontSize": 14,
          "fontStyle": "italic",
          "minHeight": 120,
          "paddingBottom": 8,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 8,
        },
        Array [
          Object {
            "marginBottom": 20,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 20,
            "paddingTop": 22,
          },
          undefined,
        ],
      ]
    }
    value="test word"
  />
</View>
`;
